lib/dataset/sort.cpp

```diff
@@ -59,16 +59,25 @@ Variable indices_for_sorting(const Variable &key, const SortOrder order) {
       core::time_point>::apply<IndicesForSorting>(key.dtype(), key, order);
 }
 
+void require_same_shape(const Dimensions &a, const Dimensions &b,
+                        const Dim dim) {
+  if (a[dim] != b[dim])
+    throw except::DimensionError(
+        "Cannot sort based on key with different shape.");
```

Comment:
I think shape can be different (the sorted data may be 2d?). Refer to bin-edges instead?

Reply:
Ok about not using just 'shape', but it's not only bin edges. The following is also an issue:
```Py
sc.sort(sc.arange('x', 10.), key=sc.arange('x', 50.))
```
gives
```
<scipp.Variable> (x: 50)    float64  [dimensionless]  [0, 1, ..., 8.18433e-85, 4.56532e-85]
```

Reply:
Then say "the length of the sort key"?

Reply:
See the latest commits?